### PR TITLE
Feat/ec 39 profil setting page

### PR DESCRIPTION
**Pull request by PaulineHavez**
**About settings page :** 

- Adding of categories filter & fix design of favorites cards

- Adding toast and logout if there is a modification of password

_--> Modifications of place.args & place mock Factory to change the maxLength of description's place_

**Do you need to restart Database ?** Yes

### CODE REVIEW

**Review by VinchC — APPROVED** [review]

File: web-app/src/components/settings/CategoriesFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  setSelectedCategories: React.Dispatch<React.SetStateAction<string[]>>;
+}
+
+export default function CategoriesFilter({
+  selectedCategories,
+  setSelectedCategories,
+}: CategoriesFilterProps) {
+  const [showCategoriesMenu, setShowCategoriesMenu] = useState(false);
+  const [categories, setCategories] = useState<Partial<Category>[]>([]);
+  const { data, loading, refetch } = useQuery(GET_CATEGORIES);
```
Comment (JonasJallet) [inline web-app/src/components/settings/CategoriesFilter.tsx:18]: Please type this useQuery with the good graphql type

**Review by JonasJallet — CHANGES_REQUESTED** [review]
It's beautiful, just please fix the little details

File: web-app/src/components/settings/FavoriteCard.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  RemoveFavorite: (idPlace: string) => Promise<void>;
+}
+
+export default function FavoriteCard({ favorite, RemoveFavorite }: Props) {
+  const [clickedFavorite, setClickedFavorite] = useState(false);
+  const router = useRouter();
+
+  const { place, setPlace } = useContext(PlaceContext) as PlaceContextType;
+
+  const handleMapClick = (favorite: Place) => {
```
Comment (JohannaStoicanescu) [inline web-app/src/components/settings/FavoriteCard.tsx:18]: something like handleDisplayFavoriteOnMap would be more clear

File: web-app/src/components/settings/FavoriteCard.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import PlaceContext from "@/contexts/PlaceContext";
+import { PlaceContextType } from "@/contexts/PlaceContext";
+import { Place } from "@/gql/generate/graphql";
+
+interface Props {
+  favorite: Place;
+  RemoveFavorite: (idPlace: string) => Promise<void>;
+}
+
+export default function FavoriteCard({ favorite, RemoveFavorite }: Props) {
```
Comment (JohannaStoicanescu) [inline web-app/src/components/settings/FavoriteCard.tsx:12]: Why is RemoveFavorite in uppercase ? try removeFromFavorite

**Review by JohannaStoicanescu — APPROVED** [review]

File: web-app/src/components/settings/FavoriteCard.tsx (hunk 3 above)
Comment (PaulineHavez) [reply web-app/src/components/settings/FavoriteCard.tsx:12]: done

File: web-app/src/components/settings/FavoriteCard.tsx (hunk 2 above)
Comment (PaulineHavez) [reply web-app/src/components/settings/FavoriteCard.tsx:18]: done

**Review by JonasJallet — APPROVED** [review]
Wonderful

**Pull request merged**